datafusion/sql/src/unparser/rewrite.rs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::sync::Arc;
+
+use datafusion_common::{
+    tree_node::{Transformed, TransformedResult, TreeNode, TreeNodeRewriter},
+    Result,
+};
+use datafusion_expr::{Expr, LogicalPlan, Sort};
+
+/// Normalize the schema of a union plan to remove qualifiers from the schema fields.
+pub(super) struct NormalizeUnionSchema {}
+
+impl NormalizeUnionSchema {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl TreeNodeRewriter for NormalizeUnionSchema {
```

Comment:
FYI because these two rewriters don't have state, you could also implement them using transform_up -- like

```rust
let plan = plan.transform_up(|plan| {
    match plan {
            LogicalPlan::Union(mut union) => {
                let schema = match Arc::try_unwrap(union.schema) {
                    Ok(inner) => inner,
                    Err(schema) => (*schema).clone(),
                };
...
})?.data
```

Reply:
TIL! Will rewrite

Reply:
I rewrote this to use `transform_up`